changing from redeem to offer (#113)
Agoric/documentation: main/zoe/api/Zoe-Intro.md

To start using a contract, you need to *install* it on Zoe. This is putting the contract’s code into Zoe. Multiple contracts can be installed on Zoe, and each contract can have multiple simultaneous active offers, each of which is handled by a separate contract instance. For clarity, in this document a *contract* is the installed code that defines how the contract works. A *contract instance* is an active execution of the contract code.

> Once installed on Zoe, a contract is instantiated, staring a lifecycle new contract instance of:

Reviewer: ```suggestion
Once installed on Zoe, a contract is instantiated, starting a lifecycle that consists of:
```

Reviewer 2: I think we shouldn't talk about contract lifecycles generally, because there isn't a general one. We could talk about the Atomic Swap lifecycle, though, and that would be informative and accurate.

1. Instantiating the contract with
   ```js
   const initialInviteP = zoe~.makeInstance(
     installationHandle,
     issuerKeywordRecord,

>      terms);

Reviewer: Our eslint settings would insist on a trailing comma and newline before the close paren. Do you want it to be relatively pretty, or to reflect our style?

Reviewer 2: We shouldn't use ~ now in our docs, since we're not yet suggesting it to users, because we don't yet have linting set up correctly in at least VSCode

   ```
1. As shown above, `makeInstance` returns a promise for the initial invite, shown above as `initialInviteP`.
1. Sometimes: obtaining the `publicAPI` from that initial invite, and then obtaining invites from that.
1. Sending out invites to request offers (including counter-offers).
1. Parties use those invites to make offers.

> 1. Determining when an offer *matches* another offer.

Reviewer 2: I think the concept of matching should be clearly restricted to Atomic Swap, so that users don't think that it is a Zoe concept.

1. Either succeeding or failing to make the trade.

An offer’s completion only affects that offer and the assets involved in it. The contract instance it was made on keeps running, waiting for and processing new offers.

> *Offer safety* is automatically enforced for parties in a contract; when an offer *completes*, it can complete giving the requesting party what they wanted, or refunding what it offered.

Reviewer: ```suggestion
*Offer safety* is automatically enforced for parties in a contract; when an offer *completes*, it can give the requesting party what they wanted, or refund what they offered.
```

> *Offer safety* is automatically enforced for parties in a contract; when an offer *completes*, it can complete giving the requesting party what they wanted, or refunding what it offered.

Reviewer 2: ```suggestion
*Offer safety* is automatically enforced for parties in a contract; when an offer *completes*, Zoe pays the party what they wanted or refunds them what they gave.
```

^ I think saying the "offer/it gives" is confusing.

> All parties in a Zoe contract instance receive an invite, which lets them make one offer to zoe for that contract instance. As a result of the offer, they receive a *promise* for a future payout. The contract itself does not have direct access to any parties’ assets. The payments for an offer are all run through Zoe, which *escrows* these assets.

Reviewer: I'm not sure that everyone who touches an invite should be called a party to it.
I think I'd say that interested parties start with an invite, and if they make an offer, they become a party and receive a promise. Or we could leave out the party.

> Zoe has two distinct APIs. One is client/user-facing zoe *service* object. We'll hold a remote promise for the zoe servive object in a variable named `zoeP` in our code examples.  The other is only exposed to contracts built on Zoe (its *contract facet*), which we'll hold in a variable named `zcf` in our code examples.

Reviewer: s/servive/service/

> Zoe has two distinct APIs. One is client/user-facing zoe *service* object. We'll hold a remote promise for the zoe servive object in a variable named `zoeP` in our code examples.  The other is only exposed to contracts built on Zoe (its *contract facet*), which we'll hold in a variable named `zcf` in our code examples.

Reviewer 2: Why is Zoe a promise and not a presence? It should be a presence, and we then don't need to call it `zoeP` - we can reference the calls like `E(zoe).makeInstance().`

Author: That's a good point. `E(zoe)` it is.

## Starting a contract on Zoe

> While there can be more than one Zoe instance installed and running at any time, for now we’ll focus on the single instance running on Agoric’s testnet. Of course, before you can run a smart contract on Zoe, you have to install it there. For Zoe purposes, *installation* means installing the code your contract is written in (JavaScript and of course the ERTP and Zoe API commands we’ve defined) on a Zoe instance.

Reviewer: I wouldn't say 'you have to install it there'. I could go with 'you have to install it.' or 'you have to tell zoe about the contract' if you want to be less formal.

I would also reword 'installing the code your contract is written in'

```suggestion
While there can be more than one Zoe instance installed and running at any time, for now we’ll focus on the single instance running on Agoric’s testnet. Of course, before you can run a smart contract on Zoe, you have to install the code. For Zoe purposes, *installation* means providing the code for your contract (in JavaScript and of course using the ERTP and Zoe API commands we’ve defined) to a Zoe instance.
```

> Contract installation returns an `installationHandle`, a unique token identifying the installed contract, used to create *contract instances* from the installed code. A contract installation is a way to reuse the same code multiple times and ensure that widely used contract code is recognizable under the same identity. For example, if I negotiate with someone and we agree to use the “Atomicswap” contract **tyg todo link**, if I make an instance of that contract, my counterparty can use its `installationHandle` with the `getInstallation()` API method to check the code is the `Atomicswap` code they expected.

Reviewer: s/a way to reuse the same code/a way to reuse known code/

First, use `zoe.makeInstance()` to create an instance of the installed contract you want to use. The method takes as arguments:
- the `installationHandle` of the specific contract to use.
- a *keyword record* of name : value pairs.
  - The names are words used in the contract, such as “Asset” or “Bid” that, in the contract, take an amount as its value.
  - The values are references to an *issuer* (see Introduction to ERTP **tyg todo: add link**) of the type you want the associated name to accept as its value in the contract instance. For example, if your contract uses the name “Asset” for what the offer in the contract is (e.g. 3) and the resource unit is quatloos, and uses “Price” for the counter-offer, which is in moola, the keyword record value would be ((“Asset” : quatlooIssuer) (“Price” : moolaIssuer)).
- *terms*: Another set of key : value pairs.

>   -   These customize contract aspects expressed in the contract without specific values. For example, an auction contract’s may not define what the minimum bid has to be or the minimum amount one has to raise from the current high bid, just that there are such values which the contract uses. These values are different for a collection of used Harry Potter paperbacks and a signed first British hardback edition of *Harry Potter and the Philosopher’s Stone*. If you write code in the contract to access that instance’s `terms`’ name:value pairs of `('MinBid' : 25, 'Currency : 'USDollar', 'MinRaise' : 5)` or  `('MinBid' : 100000, 'Currency : 'USDollar', 'MinRaise' : 5000)`, you’ve customized the generic contract for that specific instance.

Reviewer: ```suggestion
  -   These customize contract aspects expressed in the contract without specific values. For example, an auction contract may not define what the minimum bid has to be or the minimum increment above the current high bid, just that there are such values which the contract uses. These values are different for a collection of used Harry Potter paperbacks and a signed first British hardback edition of *Harry Potter and the Philosopher’s Stone*. If you write code in the contract to access that instance’s `terms`’ name:value pairs of `('MinBid' : 25, 'Currency : 'USDollar', 'MinRaise' : 5)` or  `('MinBid' : 100000, 'Currency : 'USDollar', 'MinRaise' : 5000)`, you’ve customized the generic contract for that specific instance.
```
